pkgs/desktops/cinnamon/xapps/default.nix

```diff
@@ -0,0 +1,76 @@
+{ fetchFromGitHub
+, fetchpatch
+, glib
+, gobject-introspection
+, gtk3
+, libgnomekbd
+, libxklavier
+, meson
+, ninja
+, pkgconfig
+, python3
+, python3Packages
+, stdenv
+, vala
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xapps";
+  version = "1.6.4";
+  enableParallelBuilding = true;
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = pname;
+    rev = version;
+    sha256 = "08c8j7wv56zf8mfgjfhd0wc1bbrw0dj1lmlawvsxwfajnnyjsm0d";
+  };
+
+  NIX_CFLAGS_COMPILE = [ "-I${glib.dev}/include/gio-unix-2.0" ]; # TODO: https://github.com/NixOS/nixpkgs/issues/36468
+
+  patches = [
+    (fetchpatch { # details see https://github.com/linuxmint/xapps/pull/65
+      url = "https://github.com/linuxmint/xapps/pull/65.diff";
+      sha256 = "1imcf6nfa1ky0nkirfim3fa7bx72w3k5a5p66bigcm14n0lqli7w";
+    })
+  ];
```

Comment:
use commit, not pull diffs:

```
hash mismatch in fixed-output derivation '/nix/store/l8nj3dvm41z1j82lsil3h39l51vhmszj-65.diff':
  wanted: sha256:1imcf6nfa1ky0nkirfim3fa7bx72w3k5a5p66bigcm14n0lqli7w
  got:    sha256:0sffclamvjas8ad57kxrg0vrgrd95xsk0xdl53dc3yivpxkfxrnk
cannot build derivation '/nix/store/j9w40qcxr63af1292m52cyz6j453whpz-xapps-1.6.4.drv': 1 dependencies couldn't be built
```

Reply:
note that fetchpatch normalizes patch content so tools like `nix-prefetch-url` will never return the proper hash.